.github/workflows/tests.yml

```diff
@@ -29,9 +29,11 @@ jobs:
       - uses: actions/cache@v3
         with:
           path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-            target
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
           key: ${{ runner.os }}-cargo-clippy
```

Comment:
Is this static key intentional? Shouldn't we use a hash of Cargo.lock or something similar to ensure the cache becomes stale correctly?

something like:
` key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}`

Reply:
We were not using a lock here because it's a library (mostly): https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries
Though you could argue that tap_aggregator being a binary, it should have a lock.

In any case, I think the biggest problem here that I didn't think about is that the cache will cache the lock (regardless of what we're doing in gitignore).

Reply:
> In any case, I think the biggest problem here that I didn't think about is that the cache will cache the lock (regardless of what we're doing in gitignore).

Actually nevermind, the lock file is in not in the cache paths anyway

Reply:
Isn't it possible for the cached data to go stale if we are using `${{ runner.os }}-cargo-clippy`? Seems like that key will always cause a cache hit, unless runners are changing OS. I would think the key would be tied to the stuff we are caching so it will get a cache miss when there is an update.

Reply:
When the lock is missing, Cargo checks for new versions of packages and downloads everything again (unless the version it wants to download is already in cache). I don't think there will be issues there.